### PR TITLE
Convert generate.sh to a python shell script.

### PR DESCRIPTION
**Pull request by litghost**
The previous "generate.sh" didn't properly handle the case where failure
arose during the invocation.  The python version has robust error
handling and logging in the event that a particular step fails.

### CODE REVIEW

**Review by mithro — APPROVED** [review]
Replacing shell scripts with Python scripts SGTM!

File: xc/xc7/tests/soc/ibex/generate.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+"""Generates sources for the Ibex example and copies them to the example build directory"""
```
[the remark this answers is not in the document]
Comment (litghost) [reply xc/xc7/tests/soc/ibex/generate.py:11]: Fixed.

File: xc/xc7/tests/soc/ibex/generate.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    except:
+        with open(log_file) as f_log:
+            for l in f_log:
+                print(l.strip(), file=sys.stderr)
+
+        raise
+
+
+if __name__ == "__main__":
+    main()
```
[the remark this answers is not in the document]
Comment (litghost) [reply xc/xc7/tests/soc/ibex/generate.py:142]: That seems unnecessary, as the new script is depending on exceptions for error handled.

**Comment by rw1nkler** [comment]
LGTM! Definitely the script is more robust than before.

**Review by rw1nkler — APPROVED** [review]

**Pull request merged**